godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -251,233 +251,419 @@ macro_rules! impl_vector_index {
+                (to - self).normalized()
+            }
+
+            /// Returns the squared distance between this vector and `to`.
+            ///
+            /// This method runs faster than [`Self::distance_to`], so prefer it if you need to compare vectors or need the squared distance for some formula.
+            #[inline]
+            pub fn distance_squared_to(self, to: Self) -> real {
+                (to - self).length_squared()
+            }
+
+            /// Returns the distance between this vector and `to`.
+            #[inline]
+            pub fn distance_to(self, to: Self) -> real {
+                (to - self).length()
+            }
 
-                Self::new($($comp),*)
+            /// Returns the dot product of this vector and `with`.
+            #[inline]
+            pub fn dot(self, with: Self) -> real {
+                self.to_glam().dot(with.to_glam())
             }
 
+            /// Returns a new vector with all components rounded down (towards negative infinity).
+            #[inline]
+            pub fn floor(self) -> Self {
+                Self::from_glam(self.to_glam().floor())
+            }
+
+            /// Returns true if each component of this vector is finite.
+            #[inline]
+            pub fn is_finite(self) -> bool {
+                self.to_glam().is_finite()
+            }
+
+            /// Returns `true` if the vector is normalized, i.e. its length is approximately equal to 1.
+            #[inline]
+            pub fn is_normalized(self) -> bool {
+                self.to_glam().is_normalized()
+            }
+
+            /// Returns `true` if this vector's values are approximately zero.
+            /// This method is faster than using [`Self::is_equal_approx`] with one value as a zero vector.
+            #[inline]
             pub fn is_zero_approx(self) -> bool {
-                $(self.$comp.is_zero_approx())&&*
+                $( self.$comp.is_zero_approx() )&&*
             }
 
-            pub fn posmod(self, pmod: $Scalar) -> Self {
+            /// Returns the result of the linear interpolation between this vector and `to` by amount `weight`.
+            /// `weight` is on the range of `0.0` to `1.0`, representing the amount of interpolation.
+            #[inline]
+            pub fn lerp(self, other: Self, weight: real) -> Self {
+                Self::new(
+                    $( self.$comp.lerp(other.$comp, weight) ),*
+                )
+            }
+
+            /// Returns the vector scaled to unit length. Equivalent to `self / self.length()`. See
+            /// also `is_normalized()`.
+            ///
+            /// If the vector is zero, the result is also zero.
+            #[inline]
+            pub fn normalized(self) -> Self {
+                // Copy Godot's implementation since it's faster than using glam's normalize_or_zero().
+                if self == Self::ZERO {
+                    return self;
+                }
```

Comment:
The first thing this method mentions is "Returns the vector scaled to unit length" and the second "Equivalent to `self / self.length()`". Both are wrong with this condition :wink: 

This method should instead use 
```rs
assert_ne!(self, Self::ZERO, "normalized() called on zero vector");
```
and document the panic.

We can later add specialized methods like `normalized_or_zero`, `normalized_or_right` etc.

Reply:
>Both are wrong with this condition 

Yes, but the next line clarifies it's behavior. This also mimics the behavior of [Godot's implementation](https://github.com/godotengine/godot/blob/master/core/math/vector3.h#L492).
>`/// If the vector is zero, the result is also zero.`

Reply:
It's dangerous because if people read docs at all, they may not read the whole, so the brief summary shouldn't contradict the body.

Regarding deviating from Godot, this is deliberate, see also [earlier discussions on the topic](https://github.com/godot-rust/gdext/pull/71#discussion_r1072196814). Having a NaN vector is almost never what you want.